### PR TITLE
Load font css from https: when required

### PR DESCRIPTION
**Pull request by eksperimental**
This fixes the issue with loading a css file from unsecure
source when the docs are loaded from https://

### CODE REVIEW

File: lib/ex_doc/formatter/html/templates/head_template.eex [hunk 1]
```diff
@@ -16,8 +16,12 @@
     <% end %>
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta name="generator" content="ExDoc v<%= ExDoc.version %>">
-    <link href="http://fonts.googleapis.com/css?family=Lato:400,300,700,900|Merriweather:300italic,300,700,700italic|Inconsolata:400,700" rel="stylesheet" type="text/css">
-    <link rel="stylesheet" href="dist/app.css" />
+    <script>
+      var cdn_protocol;
```
Comment (eksperimental) [inline lib/ex_doc/formatter/html/templates/head_template.eex]: @milmazz  @Dignifiedquire is it alright creating this global var, if not just let me know what is the best way to do it.

**Comment by josevalim** [comment]
:+1:! Let's wait for folks to validate the JS.

**Comment by Dignifiedquire** [comment]
Sorry but this makes no sense. `//` already is a shortcut to say ise `http` or `https` depending on the currently used protocol.

**Comment by Dignifiedquire** [comment]
so no need for any js here.

**Comment by eksperimental** [comment]
yes, I thought of that @Dignifiedquire this deals with the special case of file://
sorry I forgot to mention it in the commit.

**Comment by eksperimental** [comment]
@Dignifiedquire can you give me your input on the javascript?
I will rebase it cand check this one tomorrow.
I think we still need it, because I don't think it's necessary to download fonts via https when loading the page from http. (it's more traffic bc it's encrypted, right?)

**Comment by josevalim** [comment]
It seems we don't need this because we ship the fonts?

**Comment by Dignifiedquire** [comment]
We do not ship the fonts only the icons. But it is still not needed as we now always load via https and there is no good reason to not use https at any point.

**Comment by eksperimental** [comment]
I don't agree with that, 
serving the fonts over https on an http:/ file:  page adds an overhead in time (encrypting/decrypting) and data transfer (an these fonts are a couple hundread kb) for no real benefit.

**Comment by josevalim** [comment]
So please reopen and rebase so we can get this in. :D

**Pull request closed without merging**